db/osqlsession.c

```diff
@@ -354,6 +354,11 @@ int osql_sess_rcvop_socket(osql_sess_t *sess, int type, void *data, int datalen,
         if (debug_switch_test_sync_osql_cancel())
             poll(NULL, 0, 1000);
         osql_comm_signal_sqlthr_rc(&sess->target, sess->rqid, sess->uuid, 0, &sess->xerr, NULL, 0);
+        /* No an error. But since the transaction is cancelled inline without
```

Comment:
Lets keep this free in one place, where it is created (handle_sockbplog_request_session); we should return an error code that doesn't spew in the caller (not 0, which leaks the session).

Reply:
Sure, will do.

Reply:
I took a slightly different approach here. The rcode can come from sbuf2, bdb and berkdb, and sometimes it can even be a system error. It's therefore difficult to hand-pick an error code which isn't from any of the subsystems, and nearly impossibly to guarantee that future changes won't introduce a clashing rcode. So I added a new `is_cancelled` flag in the `osql_sess` structure. `handle_sockbplog_request_session()` checks the flag to determine whether it needs to clean the session up. Thoughts?